### ISSUE TITLE
Use the standard Logback Encoder interface instead of custom KafkaMessageEncoder

### ISSUE THREAD

**Issue by gquintana** [danielwegener/logback-kafka-appender#51]
I'd like to plug https://github.com/logstash/logstash-logback-encoder into the KafkaAppender to send JSON formatted logs.

As a workaround, I just wrote an adapter:
```
public class LogbackKafkaEncoder<E> implements KafkaMessageEncoder<E> {
    private Encoder<E> encoder;

    public Encoder<E> getEncoder() {
        return encoder;
    }

    public void setEncoder(Encoder<E> encoder) {
        this.encoder = encoder;
    }

    @Override
    public byte[] doEncode(E event) {
        return encoder.encode(event);
    }
}
```
and 
```
    <appender name="kafka" class="com.github.danielwegener.logback.kafka.KafkaAppender">
        <producerConfig>bootstrap.servers=localhost:9092</producerConfig>
        <topic>logstash</topic>
        <encoder class="fr.xxx.LogbackKafkaEncoder">
            <encoder class="net.logstash.logback.encoder.LogstashEncoder"/>
        </encoder>
    </appender>
```

**Comment by danielwegener** [comment]
Agreed. We should support logback 1.2+ encoders. This is however not logback <1.2 compatible (maybe there is way to ensure that only logback 1.2+ are passed at runtime, without failing with obscure exceptions). 
Or we roll (which would also upgrade to recent kafka versions) and make the use of logback 1.2+ mandatory.

TODO:
[] Upgrade to logback > 1.2
[] Remove KafkaMessageEncoder and replace it with the logback encoder interface
[] Adjust configuration examples
[] Adjust documentation

**Comment by michaelandrepearce** [comment]
@danielwegener any chance this can be tagged an pushed to maven central?

**Comment by danielwegener** [comment]
@gquintana Thanks for your feedback and patience. I think I have picked up all mentioned pain points. Could you give https://oss.sonatype.org/content/repositories/snapshots/com/github/danielwegener/logback-kafka-appender/0.2.0-SNAPSHOT/ a try?  /cc @michaelandrepearce

**Comment by michaelandrepearce** [comment]
@danielwegener have had to clone and local build as our corporate nexus only mirrors maven central tags, on local build and testing LGTM.

**Comment by gquintana** [comment]
Same as @michaelandrepearce,  LGTM

The Kafka producer raises "org.apache.kafka.common.errors.TimeoutException: Failed to update metadata after 60000 ms." which is catched in the failedDeliveryCallback when Kafka broker is not available at startup.
It may be interesting to resolve partitions synchronously when producer is initialized instead of waiting for the message to be sent:
```
        producer.partitionsFor(topic);
```
That's another issue.

**Comment by michaelandrepearce** [comment]
@danielwegener I see you added the catch clause for Timeout +1.

Any chance of a tag up in maven central?

**Comment by danielwegener** [comment]
@michaelandrepearce  I've released a 0.2.0-RC1 to maven central this morning.

**Comment by michaelandrepearce** [comment]
@danielwegener great thanks. Whats you plan to move to 0.2.0? You just waiting on a few folks to give positive feedback?

**Comment by danielwegener** [comment]
@michaelandrepearce essentially yes, although I really wanted to fix the outstanding recursive logging problem in the next bigger release.

**Comment by michaelandrepearce** [comment]
@danielwegener so we’re using the RC for a week so far without issues. So +1 feedback from me,
